lib/livebook/secrets/secret.ex

```diff
@@ -3,23 +3,27 @@ defmodule Livebook.Secrets.Secret do
   use Ecto.Schema
   import Ecto.Changeset
 
+  alias Livebook.EctoTypes.SecretOrigin
+
   @type t :: %__MODULE__{
           name: String.t(),
-          value: String.t()
+          value: String.t(),
+          origin: SecretOrigin.t()
         }
 
   @primary_key {:name, :string, autogenerate: false}
```

Comment:
In case you want to enforce all keys to be given when defining the struct:

```suggestion
  @primary_key {:name, :string, autogenerate: false}
  @enforce_keys [:name, :value, :origin]
```

Reply:
Can't enforce keys, because when calling `changeset/2`, we need to pass the empty `Secret` struct and it fails on compilation.